scipy/signal/tests/test_spectral.py

```diff
@@ -871,6 +872,18 @@ def test_phase_shifted_input(self):
         assert_allclose(f, f1)
         assert_allclose(C, C1)
 
+class TestCyclicSd:
+    def test_identical_input(self):
+        x = np.random.randn(50)
+
+        fs, nperseg = 1, 15
+        f = fftfreq(nperseg, 1/fs)
+
+        f1, _ = cyclic_sd(x, x, fs=fs, alpha=0.5, nperseg=nperseg, sym=True)
+        assert_allclose(f, f1)
+        cyclic_sd(x, x, fs=fs, alpha=0.5, nperseg=nperseg, sym=False)
+
+        assert_raises(ValueError, cyclic_sd, x, x, fs, fs)
```

Comment:
It would be better to use the new syntax here `with assert_raises(ValueError, match=......):` to make sure we are hitting the ValueError you expect

Reply:
done